app/models/school.rb

```diff
@@ -181,6 +181,31 @@ class ProcessDataError < StandardError; end
   # https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html#method-i-will_save_change_to_attribute-3F
   after_save :add_joining_observation, if: proc { saved_change_to_activation_date?(from: nil) }
 
+  def reading_date_bounds
```

Comment:
I think we'll need all of the readings, so the upper bound should be just whatever data we have.

For example: we're loading schedule data to validate 2 months of readings for a school. If we only load the temperature data for the existing readings, then we won't have temperatures for that new data.

So suggest only clipping the lower bound

Reply:
:+1:  have updated the school method